### PR TITLE
FHB-381 Fix warnings (combined)

### PR DESCRIPTION
**Pull request by lazcool**
Contains these already approved PRs:
* https://github.com/DFE-Digital/fh-services/pull/106
* https://github.com/DFE-Digital/fh-services/pull/109
* https://github.com/DFE-Digital/fh-services/pull/117
* https://github.com/DFE-Digital/fh-services/pull/112
* https://github.com/DFE-Digital/fh-services/pull/120

### CODE REVIEW

**Review by StuwiiDev — APPROVED** [review]
Approving based on all changes being approved in other pull requests

File: src/service/notification-api/src/FamilyHubs.Notification.Api/FamilyHubs.Notification.Api.csproj [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
 	<UserSecretsId>6f28d0d4-b9b3-47da-b394-8e1d17172c5c</UserSecretsId>
+	<NuGetAuditMode>direct</NuGetAuditMode>
```
Comment (AaronJY) [inline src/service/notification-api/src/FamilyHubs.Notification.Api/FamilyHubs.Notification.Api.csproj:5]: Can we add a `<TreatWarningsAsErrors>True</TreatWarningsAsErrors>` here?

File: src/service/notification-api/src/FamilyHubs.Notification.Data/FamilyHubs.Notification.Data.csproj [hunk 2]
```diff
@@ -1,5 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
+	<PropertyGroup>
+		<NuGetAuditMode>direct</NuGetAuditMode>
```
Comment (AaronJY) [inline src/service/notification-api/src/FamilyHubs.Notification.Data/FamilyHubs.Notification.Data.csproj:4]: Same here: `<TreatWarningsAsErrors>True</TreatWarningsAsErrors>`

File: src/service/notification-api/tests/FamilyHubs.Notification.UnitTests/FamilyHubs.Notification.UnitTests.csproj [hunk 3]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <IsPackable>false</IsPackable>
     <IsTestProject>true</IsTestProject>
+	<NuGetAuditMode>direct</NuGetAuditMode>
```
Comment (AaronJY) [inline src/service/notification-api/tests/FamilyHubs.Notification.UnitTests/FamilyHubs.Notification.UnitTests.csproj:6]: `<TreatWarningsAsErrors>True</TreatWarningsAsErrors>`

File: src/service/referral-api/acceptance-tests/FamilyHubs.Referral.Api.AcceptanceTests/FamilyHubs.Referral.Api.AcceptanceTests.csproj [hunk 4]
```diff
@@ -4,6 +4,7 @@
     <IsPackable>false</IsPackable>
     <IsTestProject>true</IsTestProject>
     <UserSecretsId>4c160f9a-da35-4b08-bf1b-7db4e32262c8</UserSecretsId>
+	<NuGetAuditMode>direct</NuGetAuditMode>
```
Comment (AaronJY) [inline src/service/referral-api/acceptance-tests/FamilyHubs.Referral.Api.AcceptanceTests/FamilyHubs.Referral.Api.AcceptanceTests.csproj:7]: `<TreatWarningsAsErrors>True</TreatWarningsAsErrors>`

**Review by AaronJY — CHANGES_REQUESTED** [review]
Only comment: can we add `<TreatWarningsAsErrors>True</TreatWarningsAsErrors>` to add to all csproj files? 🙌 They're missing on a few and it'd be good to enable this as standard across all of our projects

**Review by AaronJY — DISMISSED** [review]

**Review by AaronJY — APPROVED** [review]

**Pull request merged**